### PR TITLE
boards/olimexino-stm32: add support for olimexino-stm32 board

### PR DESCRIPTION
**Pull request by kmikaz51**
### Contribution description
This PR is a board support package for Olimexino-stm32 board.
It uses a STM32F103RB as CPU.

Currently, the package has theses features :
* ADC
* CAN
* I2C
* PWM
* RTC
* RTT
* SPI
* TIMER
* UART

I just need to add:
* DMA Support
* Arduino support

### Testing procedure
I tested the board using some tests presents in RIOT:
* example/hello-world
* tests/irq
* tests/leds
* tests/periph_adc
* tests/periph_gpio
* tests/periph_i2c
* tests/periph_pwm
* tests/periph_rtc
* tests/periph_rtt
* test/peripg_spi
* tests/periph_timer
* tests/periph_uart

For SPI and I2C tests, I use an Arduino Mega as interface.

### Issues/PRs references
N/A

### CODE REVIEW

File: makefiles/boards/stm32.inc.mk [hunk 1]
```diff
@@ -56,3 +56,11 @@ ifeq (dfu-util,$(PROGRAMMER))
   DFU_FLAGS ?= -a 2
   FFLAGS = -d $(DFU_USB_ID) $(DFU_FLAGS) -D $(FLASHFILE)
 endif
+
+ifeq (stm32flash,$(PROGRAMMER))
+
+	FLASHER = stm32flash
+	DEBUGGER =
+	FLASHFILE ?= $(BINFILE)
+	FFLAGS = -b 57600 -w $(FLASHFILE) -g 0x0 $(PORT_LINUX)
```
Comment (benpicco) [inline makefiles/boards/stm32.inc.mk]:
```suggestion
	PROG_BAUD ?= 57600
	FFLAGS = -b $(PROG_BAUD) -w $(FLASHFILE) -g 0x0 $(PORT)
```

File: makefiles/boards/stm32.inc.mk [hunk 2]
```diff
@@ -56,3 +56,11 @@ ifeq (dfu-util,$(PROGRAMMER))
   DFU_FLAGS ?= -a 2
   FFLAGS = -d $(DFU_USB_ID) $(DFU_FLAGS) -D $(FLASHFILE)
 endif
+
+ifeq (stm32flash,$(PROGRAMMER))
+
```
Comment (benpicco) [inline makefiles/boards/stm32.inc.mk]: drop the empty line

**Comment by benpicco** [comment]
`tests/bench_xtimer/Makefile.ci` needs to be updated too, looks good otherwise.

**Comment by benpicco** [comment]
Code looks good, please squash!
(one commit for the board, one commit for adding `stm32flash` to `smt32.inc.mk`)

File: boards/olimexino-stm32/doc.txt [hunk 3]
```diff
@@ -63,7 +63,7 @@ make BOARD=olimexino-stm32 term
 USART1 is used for flashing the board whereas USART2 is the serial Output.
 
 ## Supported Toolchains
-For using the ST Nucleo-F103RB board we strongly recommend the usage of the
+For using the OLIMEXINO STM32 board we strongly recommend the usage of the
 [GNU Tools for ARM Embedded Processors](https://launchpad.net/gcc-arm-embedded)
 toolchain.
```
Comment (benpicco) [inline boards/olimexino-stm32/doc.txt]: tbh I doubt this note is still necessary.
The STM32F103RB isn't the most recent chip, any old `arm-none-eabi-gcc` should do.

Comment (kmikaz51) [reply boards/olimexino-stm32/doc.txt]: Yes you're right. I inspired myself with the doc of the Nucleo F103RB

**Pull request merged**